x/paramfilter/test/param_filter_test.go

```diff
@@ -1,63 +1,99 @@
 package test
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/celestiaorg/celestia-app/app"
 	testutil "github.com/celestiaorg/celestia-app/test/util"
 	"github.com/celestiaorg/celestia-app/x/paramfilter"
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 	"github.com/stretchr/testify/require"
 	tmlog "github.com/tendermint/tendermint/libs/log"
-	"github.com/tendermint/tendermint/proto/tendermint/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/types"
 )
 
 func TestParamFilter(t *testing.T) {
-	app, _ := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams())
-
-	require.Greater(t, len(app.BlockedParams()), 0)
+	testApp, _ := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams())
+	require.Greater(t, len(testApp.BlockedParams()), 0)
 
 	// check that all blocked parameters are in the filter keeper
-	pph := paramfilter.NewParamBlockList(app.BlockedParams()...)
-	for _, p := range app.BlockedParams() {
+	pph := paramfilter.NewParamBlockList(testApp.BlockedParams()...)
+	for _, p := range testApp.BlockedParams() {
 		require.True(t, pph.IsBlocked(p[0], p[1]))
 	}
 
-	handler := pph.GovHandler(app.ParamsKeeper)
-	ctx := sdk.NewContext(app.CommitMultiStore(), types.Header{}, false, tmlog.NewNopLogger())
+	handler := pph.GovHandler(testApp.ParamsKeeper)
+	ctx := sdk.NewContext(testApp.CommitMultiStore(), tmproto.Header{}, false, tmlog.NewNopLogger())
 
-	for _, p := range app.BlockedParams() {
-		p := testProposal(proposal.NewParamChange(p[0], p[1], "value"))
-		err := handler(ctx, p)
+	t.Run("test that a proposal with a blocked param is rejected", func(t *testing.T) {
+		for _, p := range testApp.BlockedParams() {
+			p := testProposal(proposal.NewParamChange(p[0], p[1], "value"))
+			err := handler(ctx, p)
+			require.Error(t, err)
+			require.ErrorIs(t, err, paramfilter.ErrBlockedParameter)
+		}
+	})
+
+	t.Run("test that another proposal with a blocked param (evidence consensus params) is rejected", func(t *testing.T) {
```

Comment:
> oh true! I can remove the second test case b/c it should be covered by test case one.

In light of this, wouldn't this third (now second) test be covered by the first one as well?

Reply:
True. I think we could replace the second test case with a test that verifies that evidence params is actually in the list of `testApp.BlockedParams()...` b/c that's something we don't cover with just the first test case.

I have to modify this PR to account for v1 / v2 branching so I expect that we'll want a standalone test case to verify that evidence params don't exist in v1 and do exist in v2... but will keep this unresolved until then.